add-on/panels/js/desktopViews.jsx

```diff
@@ -0,0 +1,304 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var loop = loop || {};
+loop.shared = loop.shared || {};
+loop.shared.desktopViews = (function(_, mozL10n) {
+  "use strict";
+
+  var sharedActions = loop.shared.actions;
+  var sharedMixins = loop.shared.mixins;
+  var sharedUtils = loop.shared.utils;
+
+  var CopyLinkButton = React.createClass({
+    statics: {
+      TRIGGERED_RESET_DELAY: 2000
+    },
+
+    mixins: [React.addons.PureRenderMixin],
+
+    propTypes: {
+      callback: React.PropTypes.func,
+      dispatcher: React.PropTypes.instanceOf(loop.Dispatcher).isRequired,
+      from: React.PropTypes.string.isRequired,
```

Comment:
`from` is a bit vague here. I know its vague in the action as well, but maybe we could at least add a bit of documentation or change the prop name. How about `buttonLocation` or `locationForMetrics` or something along one of those lines?